test/test-signatures.spec.ts

```diff
@@ -86,6 +95,64 @@ describe("Signatures", function () {
       "R1A - asrt signed => error",
       testOneResponse("/invalid/response.root-unsigned.assertion-signed.xml", INVALID_SIGNATURE, 2)
     );
+    it(
+      "R1A - root signed - wantAssertionsSigned=true => error",
+      testOneResponse("/valid/response.root-signed.assertion-unsigned.xml", INVALID_SIGNATURE, 2, {
+        cert,
+        wantAssertionsSigned: true,
+      })
+    );
+    it(
+      "R1A - root unsigned - missing cert - wantAssertionsSigned=true => error",
```

Comment:
IMHO this test case and `R1A - root unsigned - asrt signed encrypted - missing cert - wantAssertionsSigned=true => error` should be augmented with code comment which states that these are "absurd" test cases because `passport-saml` should never allow situation where it consumes unsigned authentication response so that assertion is also unsigned (regardless of `WantAssertionsSigned` configuration option's value) which is exactly what `passport-saml` does when `cert` option is completely missing (or misspelled). I.e. these two cases exists just to test that at least `WantAssertionsSigned = true` situation fall to safe direction (i.e. reporting error instead of consuming  assertion).

This and `R1A - root unsigned - asrt signed encrypted - missing cert - wantAssertionsSigned=true => error` cases could be removed when `cert` is made truly mandatory configuration option ( see issues 180 and 523) and existence of `cert` option is validated during `passport-saml` setup phase.

With aforementioned code comment in place someone in the future would not have to spend time wondering whether he/she should fix these two cases somehow after `cert` is changed to mandatory value and these cases start to fail during test setup phase.

Sidenote regarding "absurdness" of allowing missing `cert` configuration and how `passport-saml` works when it doesn't have that option: For example Shibboleth IdP's documentation has following note (especially second paragraph):
>The default value of signResponses for this profile is "true", in keeping with modern best practice. As long as one of the response or assertion are signed, use of the profile is "safe" in terms of authentication integrity, but there are vulnerabilities in XML Encryption that make signing responses advisable when the most common encryption algorithms are used. Some of the backstory around the signing defaults is discussed in this thread.
>
>If you encounter a relying party that accepts an unsigned response **and** assertion that is transmitted via POST (and not artifact), you have identified an insecure implementation and should report the issue immediately while following your local security incident response process.

source: https://wiki.shibboleth.net/confluence/display/IDP4/SAML2SSOConfiguration
(exact version of that particular wiki page: https://wiki.shibboleth.net/confluence/pages/viewpage.action?pageId=55804373 )

Reply:
I've removed the wantAssertionsSigned+cert combination check to the saml initialization.
Now these two "absurd" testcases are not needed anymore because passport-saml throws at initialization.

I've removed the tests.